libs/execution/include/hpx/execution/detail/execution_parameter_callbacks.hpp

```diff
@@ -0,0 +1,27 @@
+//  Copyright (c) 2020 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/topology/topology.hpp>
+
+#include <cstddef>
+
+///////////////////////////////////////////////////////////////////////////////
+namespace hpx { namespace parallel { namespace execution { namespace detail {
+    /// \cond NOINTERNAL
+    using get_os_thread_count_type = std::function<std::size_t()>;
```

Comment:
Have you switched to used `std::function` now? Should we generally move to using it instead of our own types?

Reply:
Not really. Out of habit I've been using `std::function` whenever I've added callbacks. I'm happy to use `hpx::function_*` as well, but I'd like to do that in a separate PR so that I can clean up other instances of `std::function` as well.

Reply:
As a reminder: https://github.com/STEllAR-GROUP/hpx/issues/4672.